### PR TITLE
Add failing byte_order_mark test

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: cli/ast/mod.rs [hunk 1]
```diff
@@ -392,10 +393,15 @@ pub fn parse(
   source: &str,
   media_type: &MediaType,
 ) -> Result<ParsedModule, AnyError> {
+  let source = strip_bom(source);
```
Comment (dsherret) [inline cli/ast/mod.rs:396]: This doesn't fix the entire issue. I wonder if we should be doing this at a higher level?

Comment (ry) [reply cli/ast/mod.rs:396]: Is it V8 that is barfing on it?

Comment (dsherret) [reply cli/ast/mod.rs:396]: Yes. In `core::modules::ModuleMap::new_module` `module_request.get_specifier()`. I haven't looked into it too much.

It's strange that `cargo test` passes yet running it on the command line fails. I just did a `cargo clean && cargo build && ./target/debug/deno run --no-check cli/tests/testdata/byte_order_mark.ts` and it fails.

Comment (dsherret) [reply cli/ast/mod.rs:396]: Wow, that was weird. I looked into this again just now and ran:

```ts
./target/debug/deno run cli/tests/testdata/byte_order_mark.ts
```

...then it suddenly started working again next time I provided `--no-check`. Caching issue? Anyway, this should be good to go.

**Review by dsherret — APPROVED** [review]

**Pull request merged**